registry/config.go

```diff
@@ -99,8 +99,12 @@ func newServiceConfig(options ServiceOptions) *serviceConfig {
 	}
 
 	config.LoadAllowNondistributableArtifacts(options.AllowNondistributableArtifacts)
-	config.LoadMirrors(options.Mirrors)
-	config.LoadInsecureRegistries(options.InsecureRegistries)
+	if err := config.LoadMirrors(options.Mirrors); err != nil {
+		logrus.Fatalf("invalid registry mirror configurations: %s", err)
```

Comment:
This should rather return an error and `cmd/dockerd/daemon.go` should catch that err.